### DISCUSSION TITLE
Roomba iRobot530 + Berry

### DISCUSSION THREAD

**Discussion by techman83** [arendst/Tasmota#14136] [Show and tell]
So I wanted to play around with Berry Script and also needed a driver for an old Roomba I had laying around, plus a bunch of the Wireless Tag WT32C3-01N ESP32s. After poking around and a bit of a read of the SCI doco, I could get it to talk. Now technically the serial is out of spec (5v vs 3.3v), but it works pretty reliably. If it dies, I'll probably build a custom board with a voltage divider a built-in buck reg.

Driver is here: https://gist.github.com/techman83/45ed4f6de5ff4df0a2de036e33aab0f1

Seems to do what it says on the tin:
```
"iRobot530":{"State":"3","BatteryVoltage":"16838","ChargeDischarge":"55","Temperature":"33","Battery":"1618"},"TempUnit":"C"}
```
Obligatory pictures!

![2021-12-21_16-07-30](https://user-images.githubusercontent.com/3467784/146895654-1c7189ba-f411-4f97-9924-ac6e04014150.png)

![2021-12-10 12 59 23](https://user-images.githubusercontent.com/3467784/146895682-2f4178dd-2df5-4a75-a439-b9058c2155b9.jpg)

![2021-12-10 13 05 23](https://user-images.githubusercontent.com/3467784/146895664-8bbf94c4-0bfc-4745-ab9d-a78a1f3ae695.jpg)

Driver definitely could use some polish, but I can command a clean, force dock seeking, and get the stats out of it reliably.

**Comment by blakadder** [comment]
awesome!
